### DISCUSSION TITLE
Proposal: from `/pages/api` to `/server/api`

### DISCUSSION THREAD

**Discussion by danieljpgo** [vercel/next.js#42670] [Ideas]
### Introduction 
With the new edition of Next.js 13 and the `/app` folder, the `api` folder inside `pages` doesn't make much sense anymore. Therefore, we need a new location for the `api routers`.

In this proposal, the idea of ​​a `/server/api` folder to live alongside the `/app` folder is presented.

### Why:
- `/server` makes it possible to describe that everything in that folder will only run on the server.
- `/server` has more room for files/folders that should only exist and be accessed on the server, like: db connection, server env, etc.
- `/server/api` would remain consistent with how `api routers` is done today, inside /api folder.
- the community already uses this conversion to define what is defined for server only, like: [create-t3-app](https://github.com/t3-oss/create-t3-app), [trpc examples](https://github.com/trpc/trpc/tree/main/examples/next-prisma-starter/src/server), [remix examples](https://github.com/remix-run/examples/tree/main/socket.io).

### Examples

- basic

```
/...
├── /src
|  ├── /app
|  └── /server
|     └── /api
└── /...           
            

// or

/...
├── /app
├── /server
|     └── /api
└── /...           

```

- complex

```
/...
├── /src
|  ├── /app
|  |  └── /...
|  ├── /server
|  |  ├── /api
|  |  |  ├── /...
|  |  |  └── /auth
|  |  |     └── [...nextauth].ts
|  |  ├── db.ts
|  |  └── env.ts
|  └──── /style
|     └── global.css
└── /...        
```

**Comment by tgmarinho** [comment]
`src/server` or `src/api` seems nice

**Comment by ephraimduncan** [comment]
Having a `server/api/` but using it will ignore the `server` in the routes

Your url for requests will be `https://localhost:3000/api/`
